### PR TITLE
feat(openrouter): autogenerated update

### PR DESCRIPTION
**Pull request by polsala**
## Rationale
- Automated proposal from the OpenRouter generator delivering a fresh community utility.

## Why safe to merge
- Utility is isolated to `utils/<util_name>`.
- README + tests ship together (see folder contents).
- No secrets or credentials touched.

## Test Plan
- Follow the instructions in the generated README

## Links
- Generated docs and examples committed alongside this change.

## Mock Justification
- Not applicable; generator did not introduce new mocks.

### CODE REVIEW

**Comment by polsala** [comment]
Here's a consolidated review of the proposed changes:

```markdown
✅ What’s solid
*   **Clear Utility:** The `mood-emoji-mapper` provides a well-defined, simple, and useful function for mapping textual moods to emojis.
*   **Deterministic & Offline:** The design ensures the utility is fully offline and deterministic, which is excellent for reliability, performance, and ease of integration.
*   **Minimal Dependencies:** Relying solely on the Python standard library (as indicated by the empty `requirements.txt`) is a strong positive, reducing complexity and potential supply chain risks.
*   **Comprehensive Documentation:** The `README.md` is well-structured, offering clear explanations of purpose, installation, usage (both CLI and import), and design notes.
*   **Type Hinting:** The `src/mapper.py` module makes good use of type hints, improving code readability and maintainability.
*   **CLI Implementation:** The command-line interface is robustly implemented using `argparse`, providing a user-friendly way to interact with the utility.

🧪 Tests
*   **Fragile Import Pathing:** The `sys.path.append` logic in `tests/test_mapper.py` is currently incorrect for the given directory structure. `Path(__file__).resolve().parents[2]` points to `utils/nightly-mood-emoji-mapper`, and then appending `"src"` results in `utils/nightly-mood-emoji-mapper/src`, which is not where `mapper.py` resides. This will cause import errors if tests are run from the repository root or a different context.
    *   **Actionable Feedback:** Adjust the `sys.path.append` to correctly point to the `src` directory relative to the test file.
    ```python
    # In tests/test_mapper.py
    import sys
    from pathlib import Path

    # Current (incorrect):
    # ROOT = Path(__file__).resolve().parents[2]
    # sys.path.append(str(ROOT / "src"))

    # Suggested correction:
    # Assuming test_mapper.py is in '.../utils/mood-emoji-mapper/tests'
    # The 'src' directory is at '.../utils/mood-emoji-mapper/src'
    sys.path.append(str(Path(__file__).resolve().parent.parent / "src"))

    from src.mapper import map_mood_to_emoji
    ```
*   **Edge Case Coverage (Empty String):** While various mood scenarios are tested, there isn't a specific test case for an empty input string. The current implementation would likely return the default emoji, but an explicit test confirms this expected behavior.
    *   **Actionable Feedback:** Add a test case to verify the output for an empty input string.
    ```python
    # In tests/test_mapper.py
    def test_empty_string_input():
        assert map_mood_to_emoji("") == "😐"
    ```
*   **Edge Case Coverage (Case Sensitivity):** The `map_mood_to_emoji` function explicitly converts input to lowercase. A test case confirming this behavior would enhance robustness.
    *   **Actionable Feedback:** Add a test case to verify that the mapping is case-insensitive.
    ```python
    # In tests/test_mapper.py
    def test_case_insensitivity():
        assert map_mood_to_emoji("I AM HAPPY") == "😄"
        assert map_mood_to_emoji("i Am HaPpY") == "😄"
    ```

🔒 Security
*   **Minimal Attack Surface:** The utility's design inherently minimizes security risks. It operates offline, does not interact with external services, does not handle sensitive data, and has no external dependencies.
*   **Actionable Feedback:** No immediate security concerns are identified. The self-contained and deterministic nature of the utility is a strong security advantage.

🧩 Docs/DX
*   **Incorrect `README.md` Pathing:** The `Installation` section in `README.md` provides an incorrect `cd` command for navigating to the utility's directory. The path `utils/mood-emoji-mapper` is missing the `nightly-mood-emoji-mapper` segment.
    *   **Actionable Feedback:** Correct the `cd` command in `README.md` to reflect the actual nested directory structure.
    ```diff
    --- a/utils/nightly-mood-emoji-mapper/utils/mood-emoji-mapper/README.md
    +++ b/utils/nightly-mood-emoji-mapper/utils/mood-emoji-mapper/README.md
    @@ -16,7 +16,7 @@
     **Installation**
     
     ```bash
    -# From the repository root
    -cd utils/mood-emoji-mapper
    +# From the repository root, navigate to the utility directory
    +cd utils/nightly-mood-emoji-mapper/utils/mood-emoji-mapper
     python -m venv .venv
     source .venv/bin/activate
     pip install -r requirements.txt  # (empty, but kept for consistency)
    ```
*   **`requirements.txt` Clarification:** While including an empty `requirements.txt` for consistency is noted, the `pip install -r requirements.txt` command in the `README.md` might still imply that dependencies *could* be installed.
    *   **Actionable Feedback:** Consider adding a slightly more explicit note in the `README.md` next to the `pip install` command to clarify its purpose when the file is empty.
    ```diff
    --- a/utils/nightly-mood-emoji-mapper/utils/mood-emoji-mapper/README.md
    +++ b/utils/nightly-mood-emoji-mapper/utils/mood-emoji-mapper/README.md
    @@ -20,7 +20,7 @@
     cd utils/nightly-mood-emoji-mapper/utils/mood-emoji-mapper
     python -m venv .venv
     source .venv/bin/activate
    -pip install -r requirements.txt  # (empty, but kept for consistency)
    +pip install -r requirements.txt  # (empty, but included for consistent project setup, no external dependencies)
     ```

🧱 Mocks/Fakes
*   **No Mocks Required:** The utility's design as an offline, deterministic, and self-contained component means that mocks or fakes are not necessary for testing or operation. This simplifies development and testing significantly.
*   **Actionable Feedback:** The absence of a need for mocks or fakes is a positive outcome of the utility's design, indicating a well-isolated and testable component.
```
